### PR TITLE
#NGG-63|adding sourceMap for typescript - in .tmp (serve) and in prod…

### PR DESCRIPTION
**Pull request by cooperd**
…uction with debug mode

### CODE REVIEW

File: grunt-sections/transform-js.js [hunk 1]
```diff
@@ -9,9 +9,22 @@ module.exports = function (grunt, options) {
       grunt.file.write('app/scripts/reference.ts', '/// <reference path="../../reference.ts" />');
       grunt.file.write('test/reference.ts', '/// <reference path="../reference.ts" />');
       grunt.task.run('tsWithHack:copy');
+      grunt.task.run('copy:tmp');
```
Comment (shahata) [inline grunt-sections/transform-js.js]: why do we need to copy the typescript files to .tmp?

File: grunt-sections/transform-js.js [hunk 2]
```diff
@@ -9,9 +9,22 @@ module.exports = function (grunt, options) {
...
+      grunt.task.run('tsSourceMapModifier');
     }
   });
 
+  grunt.registerTask('tsSourceMapModifier', function () {
```
Comment (shahata) [inline grunt-sections/transform-js.js]: Please add a comment explaining what this method does

File: grunt-sections/transform-js.js (hunk 1 above)
Comment (cooperd) [inline grunt-sections/transform-js.js]: @shahata  - so we can debug when we do grunt serve/unit test when opening in browser

**Comment by shahata** [comment]
@aryelu and @borislit are working on similar thing. they already managed to get source maps working for `grunt serve`, but I think you guys should sync and think together on solutions. copying all .ts every time the watch is triggered is too much, we need to either have a more efficient copy or avoid the copy altogether. (I have no problem with single time copy during build, but not in watch path)

**Comment by borislit** [comment]
We got it working when debugging UTs in browser aswell.

**Pull request closed without merging**